src/main.rs

```diff
@@ -94,6 +96,25 @@ fn main() {
             app_args.isolate,
         );
 
+        if !app_args.no_context {
+            // Safe to unwrap: We just put this through an HTTP request
+            let mut clean_url = Url::parse(&final_url).unwrap();
+            clean_url.set_fragment(None);
+            // Safe to unwrap: must have a protocol and thus base 'cause we just used it.
+            clean_url.set_password(None).unwrap();
+            clean_url.set_username("").unwrap();
+            html.insert_str(
+                0,
+                &format!(
+                    "<!--- Downloaded from {} on {} using {} v{} -->\n",
```

Comment:
I believe HTML comments don't need more than two dashes in the beginning (https://html.com/tags/comment-tag/)